File: app/src/main/java/com/test/jarustest/ui/AssignmentViewModel.java
package com.test.jarustest.ui;

import android.content.Context;
import android.os.Build;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.test.jarustest.R;

import java.io.IOException;
import java.io.InputStream;
import java.nio.charset.StandardCharsets;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class AssignmentViewModel extends ViewModel {

    MutableLiveData<String> stringMutableLiveData = new MutableLiveData<>();

    public LiveData<String> getData(Context context) {
        if (stringMutableLiveData != null) {
            ExecutorService executorService = Executors.newSingleThreadExecutor();

            executorService.execute(() -> stringMutableLiveData.postValue(getJsonFromAssets(context)));

            executorService.shutdown();

        }
        return stringMutableLiveData;
    }

    private String getJsonFromAssets(Context context) {
        String outputString;
        try {
            InputStream is = context.getAssets().open(context.getString(R.string.file_name));

            int size = is.available();
            byte[] buffer = new byte[size];
            is.read(buffer);
            is.close();

            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
                outputString = new String(buffer, StandardCharsets.UTF_8);
            } else {
                outputString = new String(buffer, context.getString(R.string.format_name));
            }
        } catch (IOException e) {
            e.printStackTrace();
            return null;
        }

        return outputString;
    }
}

File: app/src/main/java/com/test/jarustest/ui/AssignmentViewModel.kt
package com.test.jarustest.ui

import android.content.Context
import android.os.Build
import androidx.annotation.RequiresApi
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.test.jarustest.R
import java.io.IOException
import java.nio.charset.Charset
import java.nio.charset.StandardCharsets
import java.util.concurrent.Executors

class AssignmentViewModel : ViewModel() {
    var stringMutableLiveData: MutableLiveData<String?>? = MutableLiveData()
    fun getData(context: Context): LiveData<String?>? {
        if (stringMutableLiveData != null) {
            val executorService = Executors.newSingleThreadExecutor()
            executorService.execute { stringMutableLiveData!!.postValue(getJsonFromAssets(context)) }
            executorService.shutdown()
        }
        return stringMutableLiveData
    }

    private fun getJsonFromAssets(context: Context): String? {
        val outputString: String
        val charset: Charset = Charsets.UTF_8

        outputString = try {
            val `is` = context.assets.open(context.getString(R.string.file_name))
            val size = `is`.available()
            val buffer = ByteArray(size)
            `is`.read(buffer)
            `is`.close()
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
                String(buffer, StandardCharsets.UTF_8)
            } else {
                String(buffer, charset)
            }

        } catch (e: IOException) {
            e.printStackTrace()
            return null
        }
        return outputString
    }
}